mne/beamformer/_lcmv.py

```diff
@@ -776,9 +795,21 @@ def tf_lcmv(epochs, forward, noise_covs, tmin, tmax, tstep, win_lengths,
                          'underlying raw object. Please use preload=False '
                          'when constructing the epochs object')
 
-    picks = _setup_picks(picks, epochs.info, forward, noise_covs[0])
+    if noise_covs is None:
+        picks = _setup_picks(picks, epochs.info, forward)
+    else:
+        picks = _setup_picks(picks, epochs.info, forward, noise_covs[0])
     ch_names = [epochs.ch_names[k] for k in picks]
 
+    # check number of sensor types present in the data
+    info_pick = pick_info(epochs.info, sel=picks)
+    ch_types =\
+        [_contains_ch_type(info_pick, tt) for tt in ('mag', 'grad', 'eeg')]
+    if sum(ch_types) > 1 and noise_covs is None:
+        raise ValueError('Source reconstruction with several sensor types '
+                         'requires a set of noise covariance matrices to be '
+                         'able to apply whitening.')
```

Comment:
please factorize the code with lcmv to avoid duplication. Use a private function eg _check_one_ch_type